app/controllers/user_journey_controller.rb

```diff
@@ -134,12 +139,10 @@ def merge_errors(primary, *objects)
   end
 
   def error_class(type)
-    if errors_present?
-      if type == 'file' && params['upload-certificate'] == 'file'
-        'govuk-file-upload--error'
-      elsif type == 'string' && params['upload-certificate'] == 'string'
-        'govuk-textarea--error'
-      end
+    if errors_present? && type == 'file' && params['upload-certificate'] == 'file'
```

Comment:
I would argue this makes it less readable, and less efficient since every time it needs to evaluate `errors_present?` twice

Reply:
fair enough, about efficiency